### PR TITLE
Add http proxy support for OIDC realm

### LINKED ISSUE #53379: Expose proxy settings for OIDC outgoing connections
Our OpenID Connect realm implementation needs to communicate on a backchannel with the OP ( in order to call the Token and UserInfo endpoints. It does so, by using an instance of `CloseableHttpAsyncClient`. `CloseableHttpAsyncClient` supports an http proxy and we should enable our realm to use one by exposing relevant configuration in the realm configuration.

### PR DESCRIPTION
**Pull request by jkakavas**
This change introduces support for using an http proxy for egress
communication of the OpenID Connect realm.

resolves: https://github.com/elastic/elasticsearch/issues/53379

### CODE REVIEW

**Review by albertzaharovits — CHANGES_REQUESTED** [review]
I think we should do a better job validating the values and expressing the dependency between the new proxy settings.
I have left suggestions with my preferred approach, but feel free to elaborate freely over them.
In addition we should add tests in `OpenIdConnectRealmSettingsTests` and also maybe update the docs to reflect the valid values (eg http/https for scheme).

I also _think_ we should support configuring ssl context parameters (eg, a different truststore, like at `HttpExporter#configureSslStrategy`), as I doubt the internal proxies to the OP have public certs.
I'm not sure though, maybe in a docker world and given that we package the runtime, it's actually decent to recommend appending the JVM trustore; but I doubt it. Nevertheless, we can tackle this in a follow-up (raising an issue for this is enough for the moment IMO).

**Comment by jkakavas** [comment]
> I also think we should support configuring ssl context parameters (eg, a different truststore, like at HttpExporter#configureSslStrategy), as I doubt the internal proxies to the OP have public certs.
I'm not sure though, maybe in a docker world and given that we package the runtime, it's actually decent to recommend appending the JVM trustore; but I doubt it. Nevertheless, we can tackle this in a follow-up (raising an issue for this is enough for the moment IMO).

We _do_ already expose TLS settings specific to the outgoing connections relevant to the OIDC realm : https://github.com/elastic/elasticsearch/blob/bbaa1f5fd5dabe7d9e2781f0eade947cbb1223fb/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/oidc/OpenIdConnectAuthenticator.java#L568

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/oidc/OpenIdConnectRealmSettings.java [hunk 1]
```diff
@@ -139,6 +139,15 @@ private OpenIdConnectRealmSettings() {
     public static final Setting.AffixSetting<Integer> HTTP_MAX_ENDPOINT_CONNECTIONS
         = Setting.affixKeySetting(RealmSettings.realmSettingPrefix(TYPE), "http.max_endpoint_connections",
         key -> Setting.intSetting(key, 200, Setting.Property.NodeScope));
+    public static final Setting.AffixSetting<String> HTTP_PROXY_HOST
+        = Setting.affixKeySetting(RealmSettings.realmSettingPrefix(TYPE), "http.proxy.host",
+        key -> Setting.simpleString(key, Setting.Property.NodeScope));
```
[the remark this answers is not in the document]
Comment (jkakavas) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/oidc/OpenIdConnectRealmSettings.java:144]: > The value validation for hostname is probably overkill in this form

Nah, I like delegating the validation to the class that would actually do it in runtime too, +1

**Comment by jkakavas** [comment]
> I think we should do a better job validating the values and expressing the dependency between the new proxy settings.

True , true, that was sloppy - will ammend

**Comment by albertzaharovits** [comment]
> We do already expose TLS settings specific to the outgoing connections relevant to the OIDC realm

Gotcha, you're right, I got confused thinking that we need a ssl context for the `.proxy*` namespace.

**Comment by jkakavas** [comment]
Ping @albertzaharovits this is ready for another round

**Comment by jkakavas** [comment]
@elasticmachine update branch

File: docs/reference/settings/security-settings.asciidoc [hunk 2]
```diff
@@ -1347,6 +1347,24 @@ id tokens with regards to their creation and expiration times.
 Specifies whether to populate the {es} user's metadata with the values that are
 provided by the OpenID Connect claims. Defaults to `true`.
 
+`http.proxy.host`::
+Specifies the address of the proxy server that will be used by the internal
+http client for all back-channel communication to the OpenID Connect Provider
```
Comment (albertzaharovits) [inline docs/reference/settings/security-settings.asciidoc:1352]: nit "backchannel" is a valid english word.

File: x-pack/qa/oidc-op-tests/build.gradle [hunk 3]
```diff
@@ -12,13 +12,15 @@ dependencies {
 }
 testFixtures.useFixture ":x-pack:test:idp-fixture", "oidc-provider"
 
-String ephemeralPort;
+String ephmeralOpPort
```
[the remark this answers is not in the document]
Comment (albertzaharovits) [reply x-pack/qa/oidc-op-tests/build.gradle:15]: this nit is still standing

**Review by albertzaharovits — APPROVED** [review]
LGTM 

Thank you for the patience, Ioannis! I should've prioritized this review more, apologies!

File: x-pack/qa/oidc-op-tests/build.gradle (hunk 3 above)
[the remark this answers is not in the document]
Comment (jkakavas) [reply x-pack/qa/oidc-op-tests/build.gradle]: :facepalm:

**Pull request merged**